main/src/com/google/refine/expr/functions/strings/ReplaceEach.java

```diff
@@ -25,13 +25,23 @@ public class ReplaceEach implements Function {
     @Override
     public Object call(Properties bindings, Object[] args) {
         if (args.length == 3) {
-            if (args[0] instanceof String && args[1] instanceof String[] && (args[2] instanceof String[] || args[2] instanceof String)) {
-                String str = (String) args[0];
-                String[] search = (String[]) args[1], replace;
+            if (args[0] instanceof String && args[1] instanceof Object[] && (args[2] instanceof Object[] || args[2] instanceof String)) {
+                String[] searchArgs = new String[((Object[]) args[1]).length];
+                Object[] originalSearchArgs = (Object[]) args[1];
+                for (var i = 0; i < searchArgs.length; i++) {
+                    searchArgs[i] = originalSearchArgs[i].toString();
+                }
+                String[] search = searchArgs, replace;
+
                 if (args[2] instanceof String) {
                     replace = new String[] { (String) args[2] };
                 } else {
-                    replace = (String[]) args[2];
+                    String[] replaceArgs = new String[((Object[]) args[2]).length];
+                    Object[] originalReplaceArgs = (Object[]) args[2];
+                    for (var i = 0; i < replaceArgs.length; i++) {
+                        replaceArgs[i] = originalReplaceArgs[i].toString();
```

Comment:
For consistency in string conversion, please use our utility method from `com.google.refine.util.StringUtils` to handle this (you'll need to adjust the imports):
```suggestion
                        replaceArgs[i] = StringUtils.toString(originalReplaceArgs[i]);
```